server/index.js

```diff
@@ -56,6 +56,97 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const sql = `
+    select "c"."cartItemId",
+            "c"."price",
+            "p"."productId",
+            "p"."image",
+            "p"."name",
+            "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using ("productId")
+    where "c"."cartId" = $1`;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(response => {
+        res.json(response.rows);
+      })
+      .catch(err => { next(err); });
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+  if (Number.isInteger(productId) && productId > 0) {
+    const sql = `
+      select "price"
+        from "products"
+        where "productId" = $1`;
+    const params = [productId];
+    db.query(sql, params)
+      .then(response => {
+        if (req.session.cartId) {
+          const cartWithPrice = {
+            price: response.rows[0].price,
+            newCartId: req.session.cartId
+          };
+          return cartWithPrice;
+        } else if (!response.rows.length) {
+          next(new ClientError(`Cannot Find Product with productId ${productId}`, 400));
```

Comment:
Since we're in a promise chain, be sure to `throw`  the error instead of calling `next()` with it.